rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -304,6 +309,12 @@ public boolean readData(String remotePath, long offset, int length, long fileSiz
     }
   }
 
+  @Override
+  public void handleHeartbeat(String workerHostname)
+  {
+    nodeManager.handleHeartbeat(workerHostname);
```

Comment:
Can you explain the flow control ? Does `RetryingBookkeeperClient.super.handleHeartbeat(workerHostname);` in `RetryingBookkeeperClient.java` trigger this call on coordinator ?